### PR TITLE
[6268] Update SDK packages

### PR DESCRIPTION
**Pull request by jvega190**
https://github.com/craftercms/craftercms/issues/6268

- Update SDK packages
- Update dependencies
- Update yarn to v4
- Migrate from CRA to Vite

### CODE REVIEW

**Comment by sumerjabri** [comment]
@jvega190 please pull and rebuild the code because deps were updated. (Be careful not to mess up the `yarn.lock`.)

**Comment by jvega190** [comment]
This is ready for merge and it's compatible with 4.1.2

File: sources/app/build.sh [hunk 1]
```diff
@@ -5,23 +5,17 @@ echo ">>> Starting app build"
...
 echo ""
 
-# Build the app
-react-scripts build
-
 echo "Deleting prior build"
 rm -rf ../../static-assets/app/*
 
+# Build the app
+vite build
```
Comment (rart) [inline sources/app/build.sh]: won't work out of IntelliJ

File: craftercms-plugin.yaml [hunk 2]
```diff
@@ -14,7 +14,7 @@ plugin:
   version:
     major: 2
     minor: 0
-    patch: 51
+    patch: 52
```
Comment (rart) [inline craftercms-plugin.yaml:17]: Must add 4.1.2 to the compatible versions

**Review by rart — CHANGES_REQUESTED** [review]
Apply to other bps

File: craftercms-plugin.yaml (hunk 2 above)
Comment (jvega190) [reply craftercms-plugin.yaml:17]: 4.1.2 is already added in all react bps (for branch 'next')

**Review by rart — APPROVED** [review]

**Pull request merged**